### ISSUE TITLE
Test Coverage Reading

### ISSUE THREAD

**Issue by jeanbrazcosta** [microsoft/vscode#209622]
Type: <b>Feature Request</b>

Good morning!

I would like to suggest an improvement regarding reading test coverage with screen readers.

Since December, test coverage has been implemented in some extensions, such as the Japa Pak extension.

However, when running the tests, I cannot know with the screen reader (I did the test with nvda), which lines are not covered, what percentage of what is covered and what would need to be covered.

To play:

Open a java project that contains unit tests, run test coverage and try to verify the results using a screen reader.

VS Code version: Code - Insiders 1.89.0-insider (d6eba9b861e3ab7d1935cff61c3943e319f5c830, 2024-04-05T05:52:23.778Z)
OS version: Windows_NT x64 10.0.22631
Modes:
Remote OS version: Linux x64 5.15.146.1-microsoft-standard-WSL2

**Comment by thgcode** [comment]
Hello!
I also want this feature.
I was researching some ways to workaround this, for some things it is possible to use the JaCoCo report directly. To map the JaCoCo colors to a description the screen reader can speak, I used this following JavaScript code pasted when opening a file with coverage data generated by JaCoCo on the browser dev tools:

```javascript
const jacocoCssClasses = new Map([[".nc", "not covered"], [".pc", "partialy covered"], [".bnc", "branch not covered"], [".bpc", "branch partialy covered"]])

for (let [cssClass, description] of jacocoCssClasses) {
	document.querySelectorAll(cssClass).forEach(line => line.ariaDescription = description);
}
```

When you apply this code to the loaded JaCoCo generated file on the browser dev tools, the code will add descriptions for the screen reader about the coverage status of the lines on the file. The lines the screen reader speaks nothing are fully covered.
Hope it helps a bit, for now until this feature is not builtin to the IDE yet.
